src/main/kotlin/io/github/gradlenexus/publishplugin/NexusPublishPlugin.kt

```diff
@@ -68,7 +82,20 @@ class NexusPublishPlugin : Plugin<Project> {
     }
 
     private fun configureNexusTasks(rootProject: Project, extension: NexusPublishExtension, registry: Provider<InvalidatingStagingRepositoryDescriptorRegistry>) {
+        rootProject.tasks.withType(AbstractNexusStagingRepositoryTask::class.java).configureEach {
```

Comment:
It just sets defaults. Gradle core plugins are built using the same convention - tasks are preconfigured whenever the plugin is applied and defaults are set.

Using `convention` means it will never override the property's value if the value was `set` somewhere else, so it avoids ordering problems.

Reply:
Makes sense. Let's merge then.